whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/NrtmVersionInfoRepository.java

```diff
@@ -22,26 +22,28 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
+import java.util.function.Function;
 
 
 @Repository
 public class NrtmVersionInfoRepository {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(NrtmVersionInfoRepository.class);
-    private final JdbcTemplate jdbcTemplate;
-    private final DateTimeProvider dateTimeProvider;
-    private final RowMapper<NrtmVersionInfo> rowMapper = (rs, rowNum) -> {
-        final NrtmSourceModel source = new NrtmSourceModel(rs.getLong(2), CIString.ciString(rs.getString(3)));
+    static final Function<Integer, RowMapper<NrtmVersionInfo>> rowMapperWithOffset = (offset) -> (rs, rowNum) -> {
```

Comment:
why do we need this offset?

Reply:
i use it in two places. with a zero it will create an NrtmVersionInfo object using all the rows returned from the query. it's also used to get DeltaVersion, where a DeltaFile object and a NrtmVersionInfo object are fetched at the same time, so then i use the offset to point to the fields for the version object.